packages/edit-post/src/components/preferences-modal/section.js

```diff
@@ -1,15 +1,24 @@
-const Section = ( { description, title, children } ) => (
-	<section className="edit-post-preferences-modal__section">
-		<h2 className="edit-post-preferences-modal__section-title">
-			{ title }
-		</h2>
-		{ description && (
-			<p className="edit-post-preferences-modal__section-description">
-				{ description }
-			</p>
-		) }
-		{ children }
-	</section>
-);
+/**
+ * WordPress dependencies
+ */
+import { renderToString } from '@wordpress/element';
+
+const Section = ( { description, title, children } ) => {
+	const content = renderToString( children );
```

Comment:
That is not great :) It's like we'll be rendering twice. Isn't there a more "data based" approach instead of using the elements?

Reply:
I know :) - it doesn't feel right to me either but it seems that we cannot use some React API to know what `children` will render, so we can't use things like `Children.count` etc..., unless there is something I'm not aware of.

Reply:
Can't we do the check in the parent and not render Section at all?

Reply:
There are lot's of checks that would be duplicate then like `PostFeaturedImageCheck`, `PostExcerptCheck` etc.. and doesn't seem good to me.

Reply:
I see what you mean but it seems this is already how we do in the document sidebar isn't it?

Reply:
I think not based on this: https://github.com/WordPress/gutenberg/blob/56f2d413b390320638b8661b5b325e43e48d834e/packages/edit-post/src/components/sidebar/settings-sidebar/index.js#L61

Did you mean something different?

The difference lies to the fact here we have a `Tab` parent. Even if I would specifically add the checks in Preferences as well, how should we handle `<PluginDocumentSettingPanel.Slot />`?

In general I don't like the current solution but couldn't find something else :)

Reply:
I mean all document sidebar panels do something like:

https://github.com/WordPress/gutenberg/blob/master/packages/edit-post/src/components/sidebar/post-author/index.js
https://github.com/WordPress/gutenberg/blob/master/packages/edit-post/src/components/sidebar/post-excerpt/index.js#L29

...

Reply:
Exactly. These checks are used in Preferences too, but as `Children` of another component `Section`. So the goal is to know if these children like `PostExcerptCheck` render something, to render the `Section` or not. The `count` will always be positive but this doesn't mean they render something.

The equivalent would be to have a check to show the `settings sidebar` or not - which we don't have.

A solution should also handle the case for `<EnablePluginDocumentSettingPanelOption.Slot />`, whether it renders something or not.

https://github.com/WordPress/gutenberg/blob/42871284be4e3b6999fc230e7501efb912d87fe7/packages/edit-post/src/components/preferences-modal/index.js#L175

Reply:
Ok I understand better now :). Unlike the panel, it seems it's not (1 check = 1 section) like I understood it initially.
I guess the proposed solution might be good enough (maybe with a comment explaining the reasoning).

Do you think we could find a CSS solution (`:empty` selector or something like that)

Curious about what others think though @gziolo @mcsf

Reply:
If you can't use [`Childrent.toArray`](https://reactjs.org/docs/react-api.html#reactchildrentoarray) that flattens the structure of children then I guess using CSS is better.

`renderToString` doesn't work with React hooks and it will quickly might become an issue.

Reply:
>Do you think we could find a CSS solution (:empty selector or something like that)

I'll try this. Thanks!

Reply:
I tried to find a css solution but with no success. We could do it with [has](https://developer.mozilla.org/en-US/docs/Web/CSS/:has) when available :).

Not really sure if this change worths to be included right now, since this seems to me like an unusual use case (to disable all panel leaving an `empty` of contents section).

Reply:
It's worth revisiting the CSS approach after 3 years to provide the simple fix at least to the supported browsers.

Reply:
I'll close the issue then 👍🏻

Reply:
You did that already, thank you Nik!